Assignment5/Assignment5_CarlosMoroteGarcia.py

```diff
@@ -0,0 +1,290 @@
+
+Additional information can be found in https://kgtk.readthedocs.io/
+
+## Simple graph statistics
+
+Let's calculate first some statistics about the KG. Count the number of instances:
+"""
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# kgtk("""
+#     query -i all
+#         --match '(instance)-[:P31]->(class)'
+#         --return 'count(distinct instance) as count_instances'
+# """)
+
+"""Now, count the number of distinct properties: 
+
+"""
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# kgtk("""
+#     query -i all
+#         --match '()-[l {label: property}]->()'
+#         --return 'count(distinct property) as count_property'
+# """)
+
+"""Now, let's count the frequency of those properties. That is, how many instances we can find with each property"""
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# kgtk("""
+#     query -i all
+#         --match '(instance)-[l {label: property}]->()'
+#         --return 'l.label, count(distinct instance) as count_instances' 
+#         --order-by 'count_instances desc'
+# """)
+
+"""## Simple queries
+Some of these queries are simple and will run in the Wikidata endpoint. 
+Try both of them using SPARQL and Kypher
+"""
+
+# Which actors has Schwarzenegger worked with throughout his career? (Print also the movie)
+
+# (in SPARQL)
+'''
+SELECT ?actor ?movie WHERE {
+  ?movie wdt:P161 wd:Q2685.
+  ?movie wdt:P161 ?actor.
+  FILTER (?actor != wd:Q2685)
+}
+'''
+
+# In Kypher:
+kgtk("""
+  query -i all
+    --match '(actor)<-[:P161]-(movie)-[:P161]->(:Q2685)'
+    --where 'actor != "Q2685"'
+    --return 'actor, movie'
+    / add-labels
+""")
+
+# How many awards does Schwarzenegger have?
+# SPARQL:
+'''
+SELECT (COUNT(?award) AS ?count_awards) WHERE {
+  wd:Q2685 wdt:P166 ?award
+}
+'''
+
+# Kypher:
+kgtk("""
+    query -i all
+        --match '(:Q2685)-[:P166]->(award)'
+        --return 'count(award) as count_awards'
+""")
+
+# Retrieve at least two members of Schwarzenegger's political party. Make sure only persons are returned
+
+# SPARQL:
+'''
+SELECT ?member ?party WHERE {
+  wd:Q2685 wdt:P102 ?party.
+  ?member wdt:P102 ?party.
+  ?member wdt:P31 wd:Q5.
+  FILTER (?member != wd:Q2685)
+}
+'''
+
+# Kypher:
+kgtk("""
+    query -i all
+        --match '(:Q2685)-[:P102]->(party)<-[:P102]-(member)-[:P31]->(:Q5)'
+        --where 'member != "Q2685"'
+        --return 'member, party'
+    / add-labels
+""")
+
+# What are the properties that describe an artist?
+
+# In theory this one is heavy on Wikidata
+
+# SPARQL: 
+'''
+SELECT DISTINCT ?property WHERE {
+  ?class rdfs:subClassOf* wd:Q483501.
+  ?element wdt:P106 ?class.
+  ?element ?property ?p.
+}
+'''
+
+
+# In Kypher:
+kgtk("""
+    query -i all
+        --match '()<-[l {label: property}]-()-[:P279*]->(:Q483501)'
```

Comment:
This is not correct. Here  you are retrieving the properties that described the artist class, instead of any instance of an artist.